doc/development_guide/profiling.rst

```diff
@@ -0,0 +1,67 @@
+.. -*- coding: utf-8 -*-
+.. _profiling:
+
+===================================
+ Profiling and performance analysis
+===================================
+
+Performance analysis for Pylint
+-------------------------------
+
+To analyse the performance of Pylint we recommend to use the ``cProfile`` module
+from ``stdlib``. Together with the ``pstats`` module this should give you all the tools
+you need to profile a Pylint run and see which functions take how long to run.
+
+The documentation for both modules can be found at cProfile_.
+
+To profile a run of Pylint over itself you can use the following code and run it from the base directory.
+It will store the profiling data in ``./profiler_stats``::
+
+    import cProfile
+    import pstats
+    import sys
+    from pstats import SortKey
+
+    sys.argv = ["pylint", "pylint"]
+    cProfile.run("from pylint import __main__", "stats")
+
+    with open("profiler_stats", "w", encoding="utf-8") as file:
+        stats = pstats.Stats("stats", stream=file)
+        stats.print_stats()
```

Comment:
Is there a call to ``stats.sort_stats()`` missing? 
You import ``SortKey`` but do not use it. It would make sense imo to sort by cumulative time.

I think the script should also delete the ``stats`` file that was created. Alternatively, the documentation should mention that it will be created, and can be loaded for interactive use if desired.

Reply:
Ah yeah, I copied the script from the more advanced script and forgot to remove the import.

I'll add a comment about `stats` being created!